src/llvm-late-gc-lowering.cpp

```diff
@@ -2468,6 +2582,31 @@ bool LateLowerGCFrame::runOnFunction(Function &F, bool *CFGModified) {
     std::map<Value *, std::pair<int, int>> CallFrames; // = OptimizeCallFrames(S, Ordering);
     PlaceRootsAndUpdateCalls(Colors, S, CallFrames);
     CleanupIR(F, &S, CFGModified);
+
+#ifdef MMTK_GC
+    // We lower the julia.gc_alloc_bytes intrinsic in this pass to insert slowpath/fastpath blocks for MMTk
+    for (BasicBlock &BB : F) {
+        for (auto it = BB.begin(); it != BB.end();) {
+            auto *CI = dyn_cast<CallInst>(&*it);
+            if (!CI) {
+                ++it;
+                continue;
+            }
+
+            Value *callee = CI->getCalledOperand();
+            assert(callee);
+
+            auto GCAllocBytes = getOrNull(jl_intrinsics::GCAllocBytes);
```

Comment:
You should be able to just iterate over users of this global rather than iterating through all statements, which is slow.